community/server/src/main/java/org/neo4j/server/web/AddressReusableServerConnector.java

```diff
@@ -0,0 +1,513 @@
+ */
+public AddressReusableServerConnector(
+    @Name("server") Server server,
+    @Name("acceptors") int acceptors,
+    @Name("selectors") int selectors,
+    @Name("sslContextFactory") SslContextFactory sslContextFactory)
+{
+    this(server,null,null,null,acceptors,selectors,AbstractConnectionFactory.getFactories(sslContextFactory,new HttpConnectionFactory()));
+}
+
+/* ------------------------------------------------------------ */
+/** Generic SSL Server Connection.
+ * @param server The {@link Server} this connector will accept connection for.
+ * @param sslContextFactory If non null, then a {@link SslConnectionFactory} is instantiated and prepended to the
+ * list of ConnectionFactories, with the first factory being the default protocol for the SslConnectionFactory.
+ * @param factories Zero or more {@link ConnectionFactory} instances used to create and configure connections.
+ */
+public AddressReusableServerConnector(
+    @Name("server") Server server,
+    @Name("sslContextFactory") SslContextFactory sslContextFactory,
+    @Name("factories") ConnectionFactory... factories)
+{
+    this(server,null,null,null,-1,-1,AbstractConnectionFactory.getFactories(sslContextFactory,factories));
+}
+
+/** Generic Server Connection.
+ * @param server
+ *          The server this connector will be accept connection for.
+ * @param executor
+ *          An executor used to run tasks for handling requests, acceptors and selectors. I
+ *          If null then use the servers executor
+ * @param scheduler
+ *          A scheduler used to schedule timeouts. If null then use the servers scheduler
+ * @param bufferPool
+ *          A ByteBuffer pool used to allocate buffers.  If null then create a private pool with default configuration.
+ * @param acceptors
+ *          the number of acceptor threads to use, or -1 for a default value. Acceptors accept new TCP/IP connections.  If 0, then
+ *          the selector threads are used to accept connections.
+ * @param selectors
+ *          the number of selector threads, or -1 for a default value. Selectors notice and schedule established connection that can make IO progress.
+ * @param factories
+ *          Zero or more {@link ConnectionFactory} instances used to create and configure connections.
+ */
+public AddressReusableServerConnector(
+    @Name("server") Server server,
+    @Name("executor") Executor executor,
+    @Name("scheduler") Scheduler scheduler,
+    @Name("bufferPool") ByteBufferPool bufferPool,
+    @Name("acceptors") int acceptors,
+    @Name("selectors") int selectors,
+    @Name("factories") ConnectionFactory... factories)
+{
+    super(server,executor,scheduler,bufferPool,acceptors,factories);
+    _manager = new ServerConnectorManager(getExecutor(), getScheduler(), selectors > 0 ? selectors : Runtime.getRuntime().availableProcessors());
+    addBean(_manager, true);
+}
+
+@Override
+protected void doStart() throws Exception
```

Comment:
Initially the best solution is to create a AddressReusableServerConnector class which extends the existing ServerConnector. 
```
AddressReusableServerConnector extends ServerConnector {
private volatile ServerSocketChannel myAcceptChannel;
private volatile int myLocalPort = -1;
// maintain _acceptChannel and _localPort field ourselves as they do not provide set method in ServerConnector class

@Override
public void open() {
//bulabula,
//change the order of two code lines.
}

// other methods related to _localPort and _acceptChannel
}
```
However this doStart method makes me no choice but to copying every thing from ServerConnector and make the class extends AbstractNetworkConnector. Because our doStart method want to do something like:
```
@Override
protected void doStart() throw Exception
{
    super.super.doStart();
    if bulabula...
}
}